server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistMeasuresStep.java

```diff
@@ -111,13 +115,33 @@ MeasureDto toMeasureDto(BatchReport.Measure in, BatchReport.Component component)
     out.setComponentId(component.getId());
     out.setSnapshotId(component.getSnapshotId());
     out.setMetricId(metricCache.get(in.getMetricKey()).getId());
-    out.setRuleId(ruleCache.get(RuleKey.parse(in.getRuleKey())).getId());
+    out.setRuleId(in.hasRuleKey() ? ruleCache.get(RuleKey.parse(in.getRuleKey())).getId() : null);
     out.setCharacteristicId(in.hasCharactericId() ? in.getCharactericId() : null);
     out.setValue(valueAsDouble(in));
     setData(in, out);
     return out;
   }
 
+  /**
+   * return the numerical value as a double. It's the type used in db.
+   * Returns null if no numerical value found
+   */
+  @CheckForNull
+  private static Double valueAsDouble(BatchReport.Measure measure) {
```

Comment:
it would be nice to keep the tests, change the visibility as package-private add @VisibleForTesting